2019/Vasilyeva/Iterator/IteratorLib/Filter/FilterDecorator.cs

```diff
@@ -0,0 +1,31 @@
+﻿using System;
+
+namespace IteratorLib
+{
+    internal class FilterDecorator<T> : IIterator<T>
+    {
+        public FilterDecorator(IIterator<T> iterator, Predicate<T> condition)
+        {
+            this.iterator = iterator;
+            this.condition = condition;
+        }
+
+        private readonly IIterator<T> iterator;
+        private readonly Predicate<T> condition;
+
+        public T Current => iterator.Current;
+
+        public bool MoveNext()
+        {
+            if(iterator.MoveNext())
+            {
+                if (condition(iterator.Current))
+                    return true;
+
+                MoveNext();
```

Comment:
I got your solution.
It's good but you missed "return".
```suggestion
                return MoveNext();
```